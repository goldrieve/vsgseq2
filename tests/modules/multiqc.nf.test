nextflow_process {

    name "Test Process MULTIQC"
    script "modules/multiqc.nf"
    process "MULTIQC"

    test("Should run multiqc on salmon quants") {

        when {
            params {
            }
            process {
                """
                input[0] = "$projectDir/tests/data/early_1_quant/"
                """
            }
        }

        then {
            assert process.success
            def outputPath = process.out['1'][0].toString()
            def statsFile = file("${outputPath}/multiqc_general_stats.txt")
            assert statsFile.exists()
            assert snapshot(statsFile).match()
        }

    }

}
