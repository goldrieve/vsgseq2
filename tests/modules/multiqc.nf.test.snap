{
    "Should run multiqc on salmon quants": {
        "content": [
            "multiqc_general_stats.txt:md5,28d27a8e4e2c279ec07df072c6e547d8"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-30T11:58:10.540607129"
    }
}